app/geant-exporter/ImportProcessConverter.cc

```diff
@@ -499,5 +534,152 @@ void ImportProcessConverter::add_table(const G4PhysicsTable* g4table,
+                = material.pdg_cutoffs.find(process_.particle_pdg)->second.energy;
+
+            auto& physics_vector
+                = element_physvec_map.find(elem_comp.element_id)->second;
+
+            for (int bin_idx : celeritas::range(energy_grid.size()))
+            {
+                const double energy_bin_value = energy_grid.at(bin_idx);
+
+                // Calculate microscopic cross-section
+                const double xs_per_atom
+                    = std::max(
+                          0.0,
+                          model.ComputeCrossSectionPerAtom(&g4_particle_def,
+                                                           &g4_element,
+                                                           energy_bin_value,
+                                                           particle_cutoff,
+                                                           energy_bin_value))
+                      / cm2;
+
+                // Store only the microscopic cross-section grid value [cm^2]
+                physics_vector.y[bin_idx] = xs_per_atom;
+            }
+        }
+
+        // Avoid cross-section vectors starting or ending with zero values.
+        // Geant4 simply uses the next/previous bin value when the vector's
+        // front/back are zero. This feels like a trap.
+        for (auto& iter : element_physvec_map)
+        {
+            auto& physics_vector = iter.second;
+
+            if (physics_vector.y.front() == 0.0)
+            {
+                // Cross-section starts with zero, use next bin value
+                physics_vector.y.front() = physics_vector.y[1];
+            }
+
+            const unsigned int last_bin = physics_vector.y.size() - 1;
+            if (physics_vector.y.back() == 0.0)
+            {
+                // Cross-section ends with zero, use previous bin value
+                physics_vector.y.back() = physics_vector.y[last_bin - 1];
+            }
+        }
+
+        // Store element cross-section map for this material
+        model_micro_xs.push_back(element_physvec_map);
+    }
+
+    return model_micro_xs;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Set up \c ImportPhysicsVector type and energy grid, while leaving
+ * cross-section data empty. This method is used to initialize physics vectors
+ * in \c this->add_micro_xs(...) .
+ *
+ * The energy grid is calculated in the same way as in
+ * \c G4VEmModel::InitialiseElementSelectors(...) .
+ */
+ImportPhysicsVector ImportProcessConverter::initialize_micro_xs_physics_vector(
+    G4VEmModel& model, const ImportMaterial& material)
+{
+    CELER_ASSERT(!material.elements.empty());
+
+    ImportPhysicsVector physics_vector;
+    physics_vector.vector_type = ImportPhysicsVectorType::log;
+
+    const double max_energy = model.HighEnergyLimit() / MeV;
+    const double min_energy
+        = material.pdg_cutoffs.find(process_.particle_pdg)->second.energy;
```

Comment:
I know we talked about this briefly on Monday, but could you remind me why we are using the production cut as the lowest energy on the grid rather than the max of the cut and model lower energy limit (like Geant4)?

Reply:
You're right. We should use the cutoffs as in Geant to keep everything consistent, but we can't be below the model's minimum energy. So that was an oversight on my side.